jina/types/document/generators.py

```diff
@@ -116,6 +118,89 @@ def from_csv(
             yield Document(value, field_resolver)
 
 
+def from_huggingface_datasets(
+    dataset_path: str,
+    field_resolver: Optional[Dict[str, str]] = None,
+    size: Optional[int] = None,
+    sampling_rate: Optional[float] = None,
+    filter_fields: Optional[bool] = False,
+    **datasets_kwargs,
+) -> Generator['Document', None, None]:
+    """Generator function for Hugging Face Datasets. Yields documents.
+
+    This function helps to load datasets from Hugging Face Datasets Hub
+    (https://huggingface.co/datasets) in Jina. Additional parameters can be
+    passed to the ``datasets`` library using keyword arguments. The ``load_dataset``
+    method from ``datasets`` library is used to load the datasets.
+
+    :param dataset_path: a valid dataset path for Hugging Face Datasets library.
+    :param field_resolver: a map from field names defined in ``document`` (JSON, dict) to the field
+            names defined in Protobuf. This is only used when the given ``document`` is
+            a JSON string or a Python dict.
+    :param size: the maximum number of the documents
+    :param sampling_rate: the sampling rate between [0, 1]
+    :param filter_fields: specifies whether to filter the dataset with the fields
+            given in ```field_resolver`` argument.
+    :param **datasets_kwargs: additional arguments for ``load_dataset`` method
+            from Datasets library. More details at
+            https://huggingface.co/docs/datasets/package_reference/loading_methods.html#datasets.load_dataset
+    :yield: documents
+    """
+    from ..document import Document
+
+    with ImportExtensions(required=True):
+        import datasets
+
+    # Validate dataset "split" argument
+    if 'split' in datasets_kwargs and datasets_kwargs['split'] is None:
```

Comment:
With the approach I've implemented right now, `split` is required. But, the point you mentioned is correct. I think a better way would be to make `split` with a default value of "all", which basically merges all the splits and returns it. I'll add a check in case user provides "None" and raise the error as defined in this flow.

Reply:
Okay, but in that case, when user gives `None` the function will fail while converting it to `Document` because `load_dataset` returns a dictionary of datasets (basically a dataset for each split, https://huggingface.co/docs/datasets/package_reference/loading_methods.html#datasets.load_dataset).

Reply:
Sure, thanks. I'll make the changes for that.

Reply:
I have modified the code with a possibly better implementation. The check is moved to post-loading the dataset with a type check. So, validation regarding the arguments to ``datasets`` library will be taken care by the library itself. On our side, we only check if the loaded dataset is in a format we can process.